file_transfer_protocol/client.py

```diff
@@ -0,0 +1,24 @@
+import socket                   # Import socket module
+
+s = socket.socket()             # Create a socket object
+host = socket.gethostname()     # Get local machine name
+port = 12312
+
+s.connect((host, port))
+s.send(b'Hello server!')
+
+with open('received_file', 'wb') as f:
+    print('file opened')
+    while True:
+        print('receiving data...')
+        data = s.recv(1024)
+        print('data=%s', (data))
+        if not data:
+            break
+        # write data to a file
+        f.write(data)
+
+f.close()
```

Comment:
__f__ is created by a __with__ context manager which automatically calls _f.close()__ so doing that again is not required.

Reply:
alright, will fix